src/app/app.component.ts

```diff
@@ -81,6 +83,7 @@ export class AppComponent implements OnInit, OnDestroy {
     this.subscribeToSettings();
     this.subscribeToIsAuthenticated();
     this.subscribeToRouterEvents();
+    this.storageService.testLocalStorage();
```

Comment:
@hhubik ok so if I understand correctly, we use `testLocalStorage();` and if there is no `localStorage` then this will throw exception but it will not kill the application because it will be handled by the `AppErrorHandler` right?

What would be also nice and can be a part of a new issue is to enhance `local-storage.service.ts` to still throw if the localStorage is missing but fall back to in memory object for the functionality for the time application runs (eg `fallbackStorage = {} as any `);

Reply:
@tomastrajan That is correct. Any exception, including one due to unavailability of `localStorage`, will cause an exception that will (1) result in a snackbar popup and (2) be logged in console. The app will not die.  I tested that as follows:
1. add `throw new Error('localStorageException');` line to `testLocalStorage` method.
2. verified that the snack bar pops up.
3. the following is in the console:
```
AppComponent_Host.ngfactory.js? [sm]:1 ERROR Error: localStorageException
    at LocalStorageService.push../src/app/core/local-storage/local-storage.service.ts.LocalStorageService.testLocalStorage (local-storage.service.ts:60)
    at AppComponent.push../src/app/app.component.ts.AppComponent.ngOnInit (app.component.ts:86)
    at checkAndUpdateDirectiveInline (core.js:18287)
    at checkAndUpdateNodeInline (core.js:19551)
    at checkAndUpdateNode (core.js:19513)
    at debugCheckAndUpdateNode (core.js:20147)
    at debugCheckDirectivesFn (core.js:20107)
    at Object.eval [as updateDirectives] (AppComponent_Host.ngfactory.js? [sm]:1)
    at Object.debugUpdateDirectives [as updateDirectives] (core.js:20099)
    at checkAndUpdateView (core.js:19495)

View_AppComponent_Host_0 | @ | AppComponent_Host.ngfactory.js? [sm]:1
-- | -- | --
  | proxyClass | @ | compiler.js:10198
  | push../node_modules/@angular/core/fesm5/core.js.DebugContext_.logError | @ | core.js:20351
  | push../node_modules/@angular/core/fesm5/core.js.ErrorHandler.handleError | @ | core.js:12326
  | push../src/app/core/error-handler/app-error-handler.service.ts.AppErrorHandler.handleError | @ | app-error-handler.service.ts:24
  | (anonymous)
    ...
```
4. After this, I can still navigate to different pages (e.g. Examples) in the app.

Reply:
@hhubik thanks for explanation!